### PR TITLE
feat: Fullstory Analytics SDK Implementation

### PR DESCRIPTION
**Pull request by HamzaIsrar12**
### Description

[LEARNER-10041](https://2u-internal.atlassian.net/browse/LEARNER-10041)

We have introduced the Fullstory Analytics Provider, which includes three main methods:

**Identify:** This method identifies the user by passing a userID (uid). Additionally, it includes a displayName for use on the Fullstory dashboard.

**Event:** This method records custom app events.

**Page:** This method functions similarly to a screen event, tracking page views.

### Config:
```
FULLSTORY:
  ENABLED: false
  ORG_ID: ''
```

### References:

- [Getting Started with Android Data Capture](https://help.fullstory.com/hc/en-us/articles/360040596093-Getting-Started-with-Android-Data-Capture)
- [Fullstory for Android - API Overview](https://developer.fullstory.com/mobile/android/)

### CODE REVIEW

**Comment by volodymyr-chekyrta** [comment]
To flag, we can merge it for now.
But it's not a widely used instrument, it will be removed from the Open edX repo and should be used as a plug-in in the future.

**Comment by omerhabib26** [comment]
Reviewing

File: dashboard/src/main/java/org/openedx/learn/presentation/LearnViewModel.kt [hunk 1]
```diff
@@ -21,4 +25,20 @@ class LearnViewModel(
...
+    fun logMyCoursesTabClickedEvent() {
+        logScreenEvent(DashboardAnalyticsEvent.MY_COURSES)
+    }
+
+    fun logMyProgramsTabClickedEvent() {
+        logScreenEvent(DashboardAnalyticsEvent.MY_PROGRAMS)
+    }
+
+    private fun logScreenEvent(event: DashboardAnalyticsEvent) {
+        analytics.logScreenEvent(event.eventName,
```
Comment (omerhabib26) [inline dashboard/src/main/java/org/openedx/learn/presentation/LearnViewModel.kt:38]: please mention parameters name

**Review by omerhabib26 — CHANGES_REQUESTED** [review]
only some minor nits,
Some test cases are failing in `VideoUnitViewModelTest.kt`

**Review by omerhabib26 — APPROVED** [review]

**Pull request merged**